packages/apps/transfer/src/pages/transfer/cross-chain-transfer-tracker/index.tsx

```diff
@@ -1,10 +1,143 @@
+import {
+  Button,
+  Heading,
+  SystemIcons,
+  TextField,
+} from '@kadena/react-components';
+
+import {
+  StyledInfoItem,
+  StyledInfoItemLine,
+  StyledInfoItemTitle,
+  StyledInfoTitle,
+} from '../cross-chain-transfer-finisher/styles';
+
 import MainLayout from '@/components/Common/Layout/MainLayout';
-import React, { FC } from 'react';
+import { useAppContext } from '@/context/app-context';
+import {
+  StyledAccountForm,
+  StyledForm,
+  StyledFormButton,
+  StyledInfoBox,
+  StyledMainContent,
+} from '@/pages/transfer/cross-chain-transfer-tracker/styles';
+import {
+  getTransferStatus,
+  IStatusData,
+} from '@/services/transfer-tracker/get-transfer-status';
+import { useRouter } from 'next/router';
+import useTranslation from 'next-translate/useTranslation';
+import React, { FC, useEffect, useState } from 'react';
 
 const CrossChainTransferTracker: FC = () => {
+  const { network } = useAppContext();
+
+  const { t } = useTranslation('common');
+  const [requestKey, setRequestKey] = useState<string>('');
+  const [data, setData] = useState<IStatusData>({});
+
+  const router = useRouter();
+
+  useEffect(() => {
+    const { reqKey } = router.query;
+    if (reqKey) {
+      setRequestKey(reqKey as string);
+    }
+  }, [router.query]);
+
+  useEffect(() => {
```

Comment:
use useDidUpdateEffect instead, since it'll call this callback on component mount which is unnecessary since the default values are the same